src/js/modules/calculator.js

```diff
@@ -0,0 +1,41 @@
+import { CALC_INPUT_TYPE } from '../constant'
+
+export default class Calculator {
+  constructor(props) {
+    this.limit = props.limit
+    this.x = ''
+    this.y = ''
+    this.operator = ''
+  }
+
+  validateInput(type) {
+    if (!this.x) return type !== CALC_INPUT_TYPE.DIGIT ? false : true
+    if (this.x.length === this.limit && !this.operator) return type !== CALC_INPUT_TYPE.OPERATION ? false : true
+    if (this.operator && !this.y) return type !== CALC_INPUT_TYPE.DIGIT ? false : true
+    if (this.y.length === this.limit) return false
+    return true
+  }
```

Comment:
```suggestion
  validateInput(type) {
    if (!this.x) return type !== CALC_INPUT_TYPE.DIGIT;
    if (this.x.length === this.limit && !this.operator) return type !== CALC_INPUT_TYPE.OPERATION;
    if (this.operator && !this.y) return type !== CALC_INPUT_TYPE.DIGIT;
    return this.y.length !== this.limit;
  }
```
boolean(true/false)를 boolean으로 변환할 필요가 있을까요?